### DISCUSSION TITLE
Add an intdiv method to nextflow.util.MemoryUnit

### DISCUSSION THREAD

**Discussion by jinmingda** [nextflow-io/nextflow#3114] [Q&A] [answered]
## New feature

Is it possible to implement an `intdiv()` method in `nextflow.util.MemoryUnit` as followed?.

```groovy
assert 7.GB == (15.GB).intdiv(2)
```

## Usage scenario 

I would like to assign a **long** type of value to memory in a process label when the memory allocation is determined programmatically based on another parameter.

```nextflow
withLabel:biTenancy {
    memory = (params.max_memory - 1.GB).intdiv(2)
}
```

## Suggest implementation 

```groovy
MemoryUnit intdiv( Number value ) {
    return new MemoryUnit( (long)(size.intdiv(value)) )
}
```

**Comment by jorgeaguileraseqera** [comment]
MemoryUnit has already a method to divide ('div' )

For example:

main.nf
```
workflow {
 def max = params.max_memory as MemoryUnit
 def mem = (max - 1.GB ) / 2
 println mem
 println mem.bytes  // long
}

```

```
nextflow run ./memunit.nf  --max_memory=10GB
N E X T F L O W  ~  version 22.04.5
Launching `./memunit.nf` [evil_mclean] DSL2 - revision: 9c0e87b85c
4.5 GB
4831838208
```

**Comment by jinmingda** [comment]
Hi @jorgeaguileraseqera thank you for your reply. To clarify, I was wondering if I could pass a long integer in GB to memory without having to converting it to bytes e.g. `assert (15.GB).intdiv(2) == 7.GB`. Currently I can only achieve this by converting memory unit to long, call intdiv and convert it back to corresponding MemorryUnit. Just thought an intdiv method directly comes with the class can be helpful in this case.

**Comment by jorgeaguileraseqera** [comment, marked as answer]
Sorry for the late response

So If I understand well you can do something as:

```
ret = MemoryUnit.of( params.input ) / 2

 println ret  // 57.3 GB
 println ret.bytes // 61561561562
 println ret.toKilo() // 60118712
```

`nextflow run ./mem.nf --input=123123123123`

**Comment by jinmingda** [comment]
@jorgeaguileraseqera sorry for the late reply. Yes that may help me solve the problem. Thank you! Please feel free to close the ticket if no further actions would be taken.
